Game/MyProject/Game/SceneManagement/Mesh.hpp

```diff
@@ -0,0 +1,41 @@
+//
+//  Mesh.hpp
+//  Game
+//
+//  Created by Tamas Both - (p) on 03/04/2019.
+//
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <IvVertexBuffer.h>
+#include <IvIndexBuffer.h>
+#include <IvStackAllocator.h>
+#include <IvRenderer.h>
+#include <IvResourceManager.h>
+
+class Mesh : std::enable_shared_from_this<Mesh> {
+private:
+    // private variable(s)
+    IvVertexBuffer* _vertex_buffer;
+    IvIndexBuffer*  _index_buffer;
+    
+    // private function(s) and method(s)
+    void destroyVertexBuffer();
+    void destroyIndexBuffer();
+    
```

Comment:
I'd much rather have public stuff before the private one, as that part is the one that people care about when they glance over your header file - the interface that they can use.